### DISCUSSION TITLE
OSError: (-30592, 'MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE') when connecting to a websocker server

### DISCUSSION THREAD

**Discussion by yamenk-gribaudo** [micropython/micropython#11721] [ESP32] [answered]
Hey everyone, I'm trying to connect to self hosted web socket server using the code below, but the connection fails with error `OSError: (-30592, 'MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE')`. If I connect to the [public socketsbay server](https://socketsbay.com/test-websockets), everything works fine, and if I use the [piesocket client](https://www.piesocket.com/websocket-tester) to connect to my server, everything also works fine. So I think the problem is a compatibility issue between me device and my server, but I'm not really know what the issue is.

MicroPython version: 1.20 from https://micropython.org/download/esp32
Board: Generic esp32 like [this one](https://micropython.org/resources/micropython-media/boards/GENERIC/esp32_devkitc.jpg)

```
import network
import time
import uwebsockets

ssid = "ssid"
password = "password"

wlan = network.WLAN(network.STA_IF)
wlan.active(True)

print("Connecting to", ssid)
wlan.connect(ssid, password)
while wlan.isconnected() == False:
    pass
print("Connected to wifi")
       
uri = "web socker server"
websocket = uwebsockets.connect(uri)
print("Connecting to {}:".format(uri))
mesg = "Some message"
websocket.send(mesg + "\r\n")
resp = websocket.recv()
print(resp)
websocket.close()
```

**Comment by davefes** [comment]
I would suggest verifying that mbedtls is fully working for ESP32.  I had problems trying to use it 6 moths ago, maybe things have changed ... although https://github.com/micropython/micropython/issues/8940 seems to suggest that there is still work being done on it.

**Reply by yamenk-gribaudo** [reply]
I think that is working fine, because if I connect the esp32 to another websocket server, all works.

**Reply by Carglglz** [reply, marked as answer]
I think you need to indicate server hostname see #9259, however `uwebsockets` does not seem to have that option, i.e. 
at https://github.com/danni/uwebsockets/blob/440edabe31f85fced35d652c18d9ac26509f8bb6/uwebsockets/client.py#L22
should be like this:
```python
...
sock = socket.socket()
addr = socket.getaddrinfo(uri.hostname, uri.port)
sock.connect(addr[0][4])
if uri.protocol == 'wss':
    sock = ussl.wrap_socket(sock, server_hostname=uri.hostname)
...
```
If this does not work I will check that the self hosted web socket server is using a compatible version of TLS i.e. up to`TLS v1.2` and that there are common cipher-suites between the server and the client, other than that it should work 👍🏼

**Reply by yamenk-gribaudo** [reply]
That was it, thank you so much

**Reply by yamenk-gribaudo** [reply]
Is there a reason why danni's library do not have that?

**Reply by baqwas** [reply]
Hate to open a new issue but I get the following error message:
`OSError: (-30592, 'MBEDTLS_ERR_SSL_FATAL_ALERT_MESSAGE')`
with one URL but not another.
Works for:
`URL = 'https://meowfacts.herokuapp.com/'`
Does not work for:
`URL = 'https://zenquotes.io/api/today/'`

My code snippet is:
```
response = urequests.get(URL)
response_code = response.status_code
```

Meowfacts returns a valid response with HTTP 200, Zenquotes does not get to returning a response.

I'm a _total newbie_ at this. My platform is:
MicroPython v1.22.1 on 2024-01-05; Arduino GIGA R1 WiFi with STM32H747

I've never used the **socket** library directly in this context (this **urequests** is my 1st foray too) but I can pre-pend header data for **urequests** if I can be given a reference link to the documentation for the library.

Thanks for the guidance.

Regards.
